static/panes/tool.ts

```diff
@@ -0,0 +1,676 @@
+
+        if (state.argsPanelShown === true) {
+            this.showPanel(this.toggleArgs, this.panelArgs);
+        }
+
+        if (state.stdinPanelShown === true) {
+            if (!this.monacoStdin) {
+                this.showPanel(this.toggleStdin, this.panelStdin);
+            } else {
+                if (!this.monacoEditorOpen) {
+                    this.openMonacoEditor();
+                }
+            }
+        }
+        this.artifactBtn.addClass('d-none');
+    }
+
+    showPanel(button: JQuery<HTMLElement>, panel: JQuery<HTMLElement>): void {
+        panel.removeClass('d-none');
+        button.addClass('active');
+        this.resize();
+    }
+
+    hidePanel(button: JQuery<HTMLElement>, panel: JQuery<HTMLElement>): void {
+        panel.addClass('d-none');
+        button.removeClass('active');
+        this.resize();
+    }
+
+    togglePanel(button: JQuery<HTMLElement>, panel: JQuery<HTMLElement>) {
+        if (panel.hasClass('d-none')) {
+            this.showPanel(button, panel);
+        } else {
+            this.hidePanel(button, panel);
+        }
+        this.saveState();
+    }
+
+    currentState() {
+        const options = this.getEffectiveOptions();
+        const state = {
+            compiler: this.compilerId,
+            editor: this.editorId,
+            tree: this.treeId,
+            wrap: options.wrap,
+            toolId: this.toolId,
+            args: this.getInputArgs(),
+            stdin: this.getInputStdin(),
+            stdinPanelShown:
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                (this.monacoStdin && this.monacoEditorOpen) || (this.panelStdin && !this.panelStdin.hasClass('d-none')),
+            monacoStdin: this.monacoStdin,
+            monacoEditorOpen: this.monacoEditorOpen,
+            monacoEditorHasBeenAutoOpened: this.monacoEditorHasBeenAutoOpened,
+            argsPanelShow: !this.panelArgs.hasClass('d-none'),
+        };
+        this.paneRenaming.addState(state);
+        this.fontScale.addState(state);
+        return state;
+    }
+
+    saveState() {
+        this.container.setState(this.currentState());
+    }
+
+    setLanguage(languageId) {
+        if (languageId) {
+            this.options.enableToggle('wrap', false);
+            const model = this.editor.getModel();
+            if (model) {
+                monaco.editor.setModelLanguage(model, languageId);
+            }
+
+            this.editor.setValue('');
+            this.fontScale.setTarget(this.editor);
+            $(this.plainContentRoot).hide();
+            $(this.editorContentRoot).show();
+        } else {
+            this.options.enableToggle('wrap', true);
+            this.plainContentRoot.empty();
+            this.fontScale.setTarget('.content');
+            $(this.editorContentRoot).hide();
+            $(this.plainContentRoot).show();
+        }
+    }
+
+    clickableUrls(text) {
```

Comment:
This could probably do with a better name while we're at it

Reply:
Renamed it to `urlToHTMLLink`